docs/advanced/local_extensions.rst

```diff
@@ -0,0 +1,60 @@
+.. _`template extensions`:
+
+Local Extensions
+----------------
+
+*New in Cookiecutter X.x*
+
+A template may extend the Cookiecutter environment with local extensions.
+These can be part of the template itself, providing it with more sophisticated custom tags and filters.
+
+To do so, a template author must specify the required extensions in ``cookiecutter.json`` as follows:
+
+.. code-block:: json
+
+    {
+        "project_slug": "Foobar",
+        "year": "{% now 'utc', '%Y' %}",
+        "_extensions": ["local_extensions.FoobarExtension"]
+    }
+
+This example assumes that a ``local_extensions`` folder (python module) exists in the template root.
+It will contain a ``main.py`` file, containing the following (for instance):
```

Comment:
Should this be updated to reflect the fact that we are adding the folder to sys path instead and the extension names would be derived from the modules/paths in the repo dir instead? The main.py may throw people off.

Reply:
I'm honestly not sure what it is you're referring to. This is meant to be an example involving a python module as a "container" for extensions, and that's literally what is being explained here.

Maybe if we're indeed going with the "local extension folder config option" idea, then maybe this example could be simplified a bit, but the basic principle is kept as simple as possible while still being pythonic, and the changes in main.py are only to facilitate the extension loading and make it all possible